src/registry.c

```diff
@@ -570,12 +570,10 @@ static inline PERSON *registry_person_get(const char *person_guid, time_t when)
         else {
             person_guid = buf;
             p = registry_person_find(person_guid);
-            if(!p) person_guid = NULL;
+            if(!p) p = registry_person_allocate(NULL, when);
```

Comment:
This will break the functionality of the registry for new users.
When a new user comes, its person_guid will be empty or null. In that case the registry allocates a new person_guid to assign to him/her.

Since you have moved the allocation inside the conditions that an person_guid already exists, it means that new users will not be able to register.